sdk/clientcore/core/src/main/java/io/clientcore/core/util/union/ParameterizedTypeImpl.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.util.union;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link ParameterizedType} implementation.
+ */
+public final class ParameterizedTypeImpl implements ParameterizedType {
```

Comment:
Why do we need this to be public?

Reply:
You can see this in the UnionTests - it is used to define generic types in the Union, e.g. 

```java
ParameterizedTypeImpl LIST_OF_STRING_TYPE
        = new ParameterizedTypeImpl(List.class, String.class);
        
Union union = Union.ofTypes(LIST_OF_STRING_TYPE, LIST_OF_INTEGER_TYPE, LIST_OF_DOUBLE_TYPE);
```

Reply:
I'll note though that when I designed this, I didn't give any thought to the right name for this type - I'm not a fan of `*Impl`

Reply:
I kind of wish the JDK had this type built-in, and we should reconcile if the client-core needs this type elsewhere as well (i.e. so it doesn't live in the `.union` package needlessly, and in which case the `union` package becomes unnecessary, and `Union` should move into `.utils` directly.)

Reply:
We should keep this as implementation and in the public APIs using this take the interface `ParameterizedType`. For the helper methods taking `Type generic, Type... args` this should be used internally.

Reply:
> We should keep this as implementation and in the public APIs using this take the interface `ParameterizedType`. For the helper methods taking `Type generic, Type... args` this should be used internally.

Do you mind explain it a little more?

For union.ofTypes(String.class, new ParameterizedTypeImpl(List.class, String.class))
We should not be able to hide the ParameterizedTypeImpl class from external user, as the ParameterizedType is a interface.

Reply:
moved this convenience class to implementation folder. We need to validate this class in TypeSpec and all other scenarios. Will introduce it back after verified.